BinaryEncoding.md

```diff
@@ -65,20 +65,40 @@ Yes:
 
 ## Global structure
 
-* A module contains:
-  * a header followed by
-  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
-  * a sequence of sections.
+* A module contains (in this order):
+  - A header, containing:
+    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
+    + Other data TBD (possibly entrypoint, memory bitness, source language, etc.)
+  - A table (sorted by offset) containing, for each section:
+    + A string literal section type name
+    + 32-bit offset within the module
```

Comment:
At Google we've exceeded the 2gb JAR size limitation in the JVM and have
had to patch the JVM. So, it could happen. And, I'm crying saying that.

On Tue, Jun 30, 2015 at 10:46 PM, Michael Holman <notifications@github.com>
wrote:

> In BinaryEncoding.md
> <https://github.com/WebAssembly/design/pull/237#discussion_r33621616>:
>
> > @@ -65,20 +65,40 @@ Yes:
> >
> >  ## Global structure
> >
> > -* A module contains:
> > -  * a header followed by
> > -  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
> > -  * a sequence of sections.
> > +* A module contains (in this order):
> > +  - A header, containing:
> > +    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
> > +    + Other data TBD (possibly entrypoint, memory bitness, source language, etc.)
> > +  - A table (sorted by offset) containing, for each section:
> > +    + A string literal section type name
> > +    + 32-bit offset within the module
>
> Yeah, I thought that those could still be 32-bit. I can see how you might
> have a bigger metadata section, but if you exhaust 32 bit address space in
> opcodes alone, then we have done something seriously wrong. And to a lesser
> extent, I have never seen a dll that was more than like 40MB, and if we are
> really a size efficient format our encoding shouldn't be orders of
> magnitude worse. A 4GB code section seems completely implausible.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/design/pull/237/files#r33621616>.
>